File: 0344 reverse-string/solution.rs
impl Solution {
    pub fn reverse_string(s: &mut Vec<char>) {
        let n=s.len();
        for i in 0..s.len()/2 {
            let temp = s[i];
            s[i]=s[n-i-1];
            s[n-i-1]=temp;
        }
    }
}

File: 0344 reverse-string/solution.java
class Solution {
    public void reverseString(char[] s) {
        rec(s,0,s.length-1);
    }
    private void rec(char[] s,int l,int r) {
        if(l>=r) return;
        char c = s[l];
        s[l]=s[r];
        s[r]=c;
        rec(s,l+1,r-1);
    }
}
